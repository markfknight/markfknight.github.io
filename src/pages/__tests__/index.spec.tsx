import React from 'react';
import { cleanup, render } from '@testing-library/react';

import Home, { Data } from '../index';
import { PageProps } from 'gatsby';

jest.mock('../../components/seo');
jest.mock('../../components/post-list');

afterEach(cleanup);

describe(`Index snapshot`, () => {
  it(`renders correclty`, () => {
    const props: PageProps<Data> = {
      location: {},
      data: {
        site: {
          siteMetadata: {
            title: 'test',
          },
        },
      },
    } as PageProps<Data>;

    const tree = render(<Home {...props} />);

    expect(tree).toMatchSnapshot();
  });
});
