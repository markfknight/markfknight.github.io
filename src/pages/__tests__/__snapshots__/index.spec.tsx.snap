// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index snapshot renders correclty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container max-w-3xl px4 py10"
      >
        <header
          class="py-10"
        >
          <h3>
            <a
              href="/"
            >
              test
            </a>
          </h3>
        </header>
        <main />
        <footer
          class="py-10 text-center"
        >
          © 
          2021
          , Built with
           
          <a
            href="https://www.gatsbyjs.org"
          >
            Gatsby
          </a>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container max-w-3xl px4 py10"
    >
      <header
        class="py-10"
      >
        <h3>
          <a
            href="/"
          >
            test
          </a>
        </h3>
      </header>
      <main />
      <footer
        class="py-10 text-center"
      >
        © 
        2021
        , Built with
         
        <a
          href="https://www.gatsbyjs.org"
        >
          Gatsby
        </a>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
